packages/remirror__extension-image/src/image-extension.ts

```diff
@@ -94,6 +96,15 @@ export class ImageExtension extends NodeExtension<ImageOptions> {
   }
 
   createTags() {
+    // I don't like this at all, would very much appreciate feedback here
+    const hasCaption = this.store.extensions.find(
+      (e: AnyExtension) => e instanceof CaptionExtension,
+    );
+
+    if (hasCaption) {
+      return [ExtensionTag.BlockNode, ExtensionTag.Media, ExtensionTag.CaptionCompatible];
+    }
+
     return [ExtensionTag.InlineNode, ExtensionTag.Media];
```

Comment:
@ifiokjr @ocavue would love some suggestions on how to improve this, I wonder if tags could be part of `nodeOverride` or something?